include/eve/module/algo/algo/search.hpp

```diff
@@ -0,0 +1,246 @@
+      //
+      // Unfortunately shuffle_v2 can't slide left yet
+      haystack = eve::slide_left(haystack, eve::index<1>);
+    }
+
+    return {};
+  }
+
+  template<relaxed_range R1, relaxed_range R2, typename P>
+  EVE_FORCEINLINE auto operator()(R1&& haystack, R2&& needle, P p) const -> unaligned_iterator_t<R1>
+  {
+    std::ptrdiff_t needle_len   = (needle.end() - needle.begin());
+    std::ptrdiff_t haystack_len = (haystack.end() - haystack.begin());
+
+    if( needle_len == 0 ) return unalign(haystack.begin());
+    if( haystack_len < needle_len ) return unalign(haystack.begin()) + haystack_len;
+
+    auto [processed_haystack, processed_needle] =
+        temporary_preprocess_ranges_hack(TraitsSupport::get_traits(), haystack, needle);
+
+    auto haystack_f = processed_haystack.begin();
+    auto haystack_l = processed_haystack.end();
+
+    using I1 = decltype(processed_haystack.begin());
+    using I2 = decltype(processed_needle.begin());
+
+    needle_checker<I1, I2, P> needle_checker(processed_needle.begin(), processed_needle.end(), p);
+
+    auto haystack_main_part_l = eve::unalign(haystack_f);
+
+    if( haystack_len > iterator_cardinal_v<I1> )
+    {
+      haystack_main_part_l += haystack_len - std::max(needle_len, iterator_cardinal_v<I1>) + 1;
+
+      if( auto res = main_part(processed_haystack.traits(),
+                               haystack_f,
+                               haystack_main_part_l,
+                               processed_needle.begin(),
+                               needle_len,
+                               p,
+                               needle_checker) )
+      {
+        return eve::unalign(haystack.begin()) + (*res - haystack_f);
+      }
+    }
+
+    if( auto res = small_tail(haystack_main_part_l, haystack_l, needle_len, needle_checker) )
+    {
+      return eve::unalign(haystack.begin()) + (*res - haystack_f);
+    }
+
+    return eve::unalign(haystack.begin()) + haystack_len;
+  }
+
+  template<relaxed_range R1, relaxed_range R2>
+  EVE_FORCEINLINE auto operator()(R1&& haystack, R2&& needle) const -> unaligned_iterator_t<R1>
+  {
+    // we should implement `common_type` as a trait for algorithms
+    // and not just zip
+    using T = common_type_t<value_type_t<R1>, eve::value_type_t<R2>>;
+
+    auto converted_res = operator()(views::convert(EVE_FWD(haystack), eve::as<T> {}),
+                                    views::convert(EVE_FWD(needle), eve::as<T> {}),
+                                    eve::is_equal);
+    return views::convert(converted_res, eve::as<value_type_t<R1>> {});
+  }
+};
+
+//=================================================================================================
+//! @addtogroup algos
+//! @{
+//!  @var search
+//!
+//!  @brief SIMD version of std::search (subsequence in a sequence).
+//!
+//!   **Defined in Header**
+//!
+//!   @code
+//!   #include <eve/module/algo.hpp>
+//!   @endcode
+//!
+//!   Some ideas taken from previous work:
```

Comment:
a bit more doc could be nice. At least repsing the std like prototype and havign a doc/ example maybe ?

Reply:
I will ofc, not finished, sorry